aten/src/ATen/LegacyTHDispatch.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+// LegacyTHDispatcher is the legacy mechanism for dispatching directly
+// to TH/THNN/THC/THCUNN functions in ATen, which is essentially a giant virtual
+// dispatch table for every TH function we support dynamically dispatching over.
```

Comment:
I think what you want to add to the comment here is:

1. Why you need another dispatcher, as opposed to using the existing LegacyTypeDispatch (you say that it's similar to the LegacyTypeDispatch with some simplifications, but that doesn't explain the "why")
2. Under what situation we can delete this (when all the TH functions go away right)